include/aws/common/allocator.h

```diff
@@ -12,6 +12,25 @@
 AWS_PUSH_SANE_WARNING_LEVEL
 AWS_EXTERN_C_BEGIN
 
+/*
+ * Quick guide to allocators:
+ *  CRT offers several flavours of allocators:
+ * - default: basic allocator, targeted toward most common use case. behavior
+ *   can change over time. Current default aligns small allocations on 8 byte
+ *   boundary and big buffers on 32/64 byte (system dependent) boundary.
+ *   Aligned mem can improve perf on some operations, like memcpy or hashes.
+ *   Depending on a system, can result in higher peak memory count in heavy
+ *   acquire/free scenarios (ex. s3), due to memory fragmentation related to how
+ *   aligned allocators work (overallocate, find aligned offset, release extra memory)
+ * - no_align: similar to default, but does not do any alignment. In practice,
```

Comment:
trivial: `no_align` name is based on the fact that our `default allocator` does alignment. What if, in the future, we change our `default allocator` to something else? Maybe something like `system_allocator`?  

I am conflicted because `no_alloc` name does signify that the reason we are using system allocator instead of default allocator is because we don't want memory allignment. So `system_allocator` might be confusing to the future reader of the codebase. Up to you.